beanstalk_worker/worker_start_scan.py

```diff
@@ -26,212 +26,340 @@
+        return json.dumps(json_data)
+
+    def run(self):
+        """"
+        Runs the listed atomic scanners on image under test
+
+        #FIXME: at the moment this menthod is returning the results of multiple
+        scanners in one json and sends over the bus
+        """
+        # find image's full name and append the desired tag
+        image_under_test = job_info.get('name').split(":")[0] + ":" \
+            + job_info.get("tag")
+
+        # pull the image first
+
+        self.pull_image_under_test(image_under_test)
+
+        scanners_data = self.job_info
+        scanners_data["action"] = "report_scan_results"
+
+        for scanner in SCANNERS.keys():
+            data_temp = self.run_a_scanner(scanner, image_under_test)
+            scanners_data["msg"] = {
+                    data_temp["scanner_name"]: data_temp["msg"]
+                    }
+            scanners_data["logs"] = {
+                    data_temp["scanner_name"]: data_temp["logs"]
+                    }
+
+        return scanners_data
+
+
+class ScannerRPMVerify(ScannerRunner):
+    """
+    scanner-rpm-verify atomic scanner handler
+    """
+    def __init__(self):
+        self.scanner_name = "scanner-rpm-verify"
+
+    def run_atomic_scanner(self, image_under_test):
+        """
+        Run the atomic scan command
+        """
+        process = subprocess.Popen([
+            "atomic",
+            "scan",
+            "--scanner=%s" % self.scanner_name,
+            "%s" % image_under_test
+        ], stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+
+        # returns out, err
+        return process.communicate()
+
+    def run(self, image_under_test):
+        """
+        Run the scanner-rpm-verify atomic scanner
+        """
+        json_data = {}
+        out, err = self.run_atomic_scanner(image_under_test)
+        if out != "":
+            output_json_file = os.path.join(
+                out.strip().split()[-1].split('.')[0],
+                self.find_image_long_id(image_under_test),
+                # TODO: provision parsing multiple output files per scanner
+                SCANNERS[self.scanner_name]["output_filename"][0]
             )
-        logger.log(
-            level=logging.INFO,
-            msg="Unmounting and removing directory %s" % image_rootfs_path)
-        mount_object.unmount()
-        shutil.rmtree(image_rootfs_path)
-        try:
-            os.makedirs(image_rootfs_path)
-        except OSError as error:
+
+            if os.path.exists(output_json_file):
+                json_data = json.loads(open(output_json_file).read())
+            else:
+                logger.log(
+                    level=logging.FATAL,
+                    msg="No scan results found at %s" % output_json_file)
+                # FIXME: handle what happens in this case
+                raise
+        else:
+            # TODO: do not exit here if one of the scanner failed to run,
+            # others might run
```

Comment:
:+1: for the TODO. As we add more scanners, we need to figure how to go about handling such a scenario. I think a good way would be to add a message saying `xyz scanner failed`. And, as we start adding more scanners, doing that sooner rather than later would make sense.